[PATCH] gstreamer-plugins-good audioparser: support SVP and 4K

When playing encrypted 4K content, Secure Video Path is required. 

On Broadcom platform 16.2 and above, a secure buffer is introduced. 
During decryption, decrypted video data is put into secure buffer 
instead of general system memory, and SVP meta data is generated and
pushed down GStreamer together with the original encrypted content.
It's up to the audio and video decoder to handle the encrypted data 
with SVP meta data.
Plugins which look into audio/video data need to be removed since 
SOC plugins already cover the audio parser functionality.

Change-Id: I27b09f0fe077d1898c4516e73c2e59144848b675
Signed-off-by: Hao Li <hao_li@cable.comcast.com>

diff --git a/gst/audioparsers/Makefile.am b/gst/audioparsers/Makefile.am
index b2b2a4d..46921d4 100644
--- a/gst/audioparsers/Makefile.am
+++ b/gst/audioparsers/Makefile.am
@@ -1,7 +1,7 @@
 plugin_LTLIBRARIES = libgstaudioparsers.la
 
 libgstaudioparsers_la_SOURCES = \
-	gstaacparse.c gstamrparse.c gstac3parse.c \
+	gstamrparse.c gstac3parse.c \
 	gstdcaparse.c gstflacparse.c gstmpegaudioparse.c \
 	gstsbcparse.c gstwavpackparse.c plugin.c
 
diff --git a/gst/audioparsers/plugin.c b/gst/audioparsers/plugin.c
index 8cbfe60..6524016 100644
--- a/gst/audioparsers/plugin.c
+++ b/gst/audioparsers/plugin.c
@@ -35,9 +35,10 @@ plugin_init (GstPlugin * plugin)
 {
   gboolean ret;
 
-  ret = gst_element_register (plugin, "aacparse",
-      GST_RANK_PRIMARY + 1, GST_TYPE_AAC_PARSE);
-  ret &= gst_element_register (plugin, "amrparse",
+  //ret = gst_element_register (plugin, "aacparse",
+  //    GST_RANK_PRIMARY + 1, GST_TYPE_AAC_PARSE);
+  //ret &= gst_element_register (plugin, "amrparse",
+  ret = gst_element_register (plugin, "amrparse",
       GST_RANK_PRIMARY + 1, GST_TYPE_AMR_PARSE);
   ret &= gst_element_register (plugin, "ac3parse",
       GST_RANK_PRIMARY + 1, GST_TYPE_AC3_PARSE);
